cypress/e2e/requests.cy.js

```diff
@@ -2,50 +2,102 @@ describe('Viewing all requests', () => {
-
       // Intercept the response from the endpoint to view all requests
+      // TODO(summer-cook): extract out this base url into the config to use as an environment variable. it was not cooperating before
       cy.intercept('GET', `${scientistApiBaseURL}/quote_groups/mine.json`, (req) => {
-        if (requestList === true) {
-          req.reply({fixture: 'all-requests/requests.json'})
-        } else {
-          req.reply({fixture: 'all-requests/no-requests.json'})
+        if ((requestList === undefined) && (loading === true)) {
+          // reply with an empty response: both data and error will be undefined.
+          req.reply()
+        } else if ((requestList === undefined) && (loading === undefined) && (error === true)) {
+          // error will be defined
+          req.reply({ statusCode: 404 })
+        } else if (requestList === true) {
+          // reply with a request body- default status code is 200
+          req.reply({ fixture: 'all-requests/requests.json' })
+        } else if (requestList === false) {
+          req.reply({ fixture: 'all-requests/no-requests.json' })
         }
-      }).as('useAllRequests')
-      cy.visit("/requests")
+      })
+      // Intercept the response from the endpoint that gets the default ware ID
+      cy.intercept('GET', `${scientistApiBaseURL}/wares.json?q=make-a-request`, { fixture: 'all-requests/make-a-request.json' })
+      cy.visit('/requests')
     })
 
-    context('has requests', () => {
+
+    context('request list is loading', () => {
       before(() => {
-        requestList = true
+        loading = true
       })
-      it("shows the user's request list.", () => {
-        cy.get('article.request-item').should('exist').then(() => {
-          cy.log('Successfully viewing request list.')
+      it('should show a loading spinner.', () => {
+        cy.get("[aria-label='tail-spin-loading']").should('be.visible').then(() => {
+          cy.log('Loading spinner displays correctly.')
         })
       })
     })
 
-    context('has 0 requests', () => {
+    context('error while making a request to the api', () => {
       before(() => {
-        requestList = false
+        requestList = undefined
+        loading = undefined
+        error = true
+      })
+      it('should show an error message.', () => {
+        cy.get("div[role='alert']").should('be.visible').then(() => {
+          cy.log('Successfully hits an error.')
+        })
       })
-      it("shows a message notifying the user they don't have any requests.", () => {
-        cy.get('p.no-requests').contains('You do not have any requests yet.').then(() => {
-          cy.log('Successfully viewing request page with no requests.')
+    })
+
+    describe('request components are loading successfully, &', () => {
+      context('the user has requests', () => {
+        before(() => {
+          requestList = true
+        })
+        it("should show the user's request list.", () => {
```

Comment:
```suggestion
        it('should show the user's request list.', () => {
```

Reply:
this causes a syntax error because of the ' in "user's".

Reply:
ah. I missed that.